src/test/notary/NotaryTest.kt

```diff
@@ -0,0 +1,107 @@
+package notary
+
+import com.nhaarman.mockito_kotlin.doReturn
+import com.nhaarman.mockito_kotlin.mock
+import io.reactivex.Observable
+import main.CONFIG
+import main.ConfigKeys
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.fail
+import org.junit.jupiter.api.Test
+import java.math.BigInteger
+
+/**
+ * Test business logic of Notary.
+ */
+class NotaryTest {
+
+    /**
+     * @given a custodian has 100 Wei with intention to deposit 100 Wei to Notary
+     * @when a custodian transfer 100 Wei to a specified wallet and specifies Iroha wallet to deposit assets
+     * @then an IrohaOrderedBatch is emitted with 3 transactions:
+     * 1 - SetAccountDetail with hash
+     * 2 - AddAssetQuantity with 100 Wei
+     * 3 - TransferAsset with 100 Wei to specified asset
+     */
+    @Test
+    fun depositEthereumTest() {
+        val amount = BigInteger.valueOf(100)
+        val accountId = "user@test"
+        val assetId = CONFIG[ConfigKeys.irohaEthToken]
+        val creatorId = CONFIG[ConfigKeys.irohaCreator]
+        val txHash = "hash"
+        val txFrom = "from"
+
+        val custodianIntention = mock<NotaryInputEvent.EthChainInputEvent.OnEthSidechainDeposit>() {
+            on { hash } doReturn txHash
+            on { from } doReturn txFrom
+            on { value } doReturn amount
+            on { input } doReturn accountId
+        }
+
+        // source of events from side chains
+        val obsEth = Observable.just<NotaryInputEvent>(custodianIntention)
+        val obsIroha = Observable.empty<NotaryInputEvent>()
+
+        val notary = NotaryImpl(obsEth, obsIroha)
+        val res = notary.irohaOutput()
+
+        res.subscribe(
+            {
+                when (it) {
+                    is IrohaOrderedBatch -> {
```

Comment:
What if not an `IrohaOrderedBatch` arrives?

Reply:
fixed